### PR TITLE
make create_wallet to wait for seed creation when using signer

### PR DESCRIPTION
**Pull request by jazz-cb**
### What changed? Why?

Make wallet creation sync, when using ServerSigner. `create_wallet` will poll CDPService to get the updated status util the wallet is ready

### CODE REVIEW

File: lib/coinbase/wallet.rb [hunk 1]
```diff
@@ -17,6 +17,17 @@ class Wallet
     # The maximum number of addresses in a Wallet.
     MAX_ADDRESSES = 20
 
+    # A representation of ServerSigner status in a Wallet.
+    module ServerSignerStatus
+      # The Wallet is awaiting seed creation by the ServerSigner. At this point,
+      # the Wallet cannot create addresses or sign transactions.
+      PENDING_SEED_CREATION = :pending_seed_creation
```
Comment (yuga-cb) [inline lib/coinbase/wallet.rb:24]: Just `PENDING`?

And do we need `FAILED`?

Comment (jazz-cb) [reply lib/coinbase/wallet.rb:24]: There is no "FAILED" state yet on the backend. A failure for the SDK will be a timeout

Comment (jazz-cb) [reply lib/coinbase/wallet.rb:24]: PENDING conflicts with the transferStatus constants. Hence renamed this to `pending_seed_creation` on backend

Comment (yuga-cb) [reply lib/coinbase/wallet.rb:24]: They are scoped in different modules so there should be no conflict

Comment (jazz-cb) [reply lib/coinbase/wallet.rb]: renamed at the SDK level

File: lib/coinbase/wallet.rb [hunk 2]
```diff
@@ -36,10 +47,14 @@ def import(data)
...
       # Creates a new Wallet on the specified Network and generate a default address for it.
+      # @param interval_seconds [Integer] The interval at which to poll the CDPService for the Wallet to
+      # have an active seed, if using a ServerSigner, in seconds
+      # @param timeout_seconds [Integer] The maximum amount of time to wait for the ServerSigner to
+      # create a seed for the Wallet, in seconds
       # @param network_id [String] (Optional) the ID of the blockchain network. Defaults to 'base-sepolia'.
       # @param server_signer [Boolean] (Optional) whether Wallet should use project's server signer. Defaults to false.
       # @return [Coinbase::Wallet] the new Wallet
-      def create(network_id: 'base-sepolia')
+      def create(interval_seconds = 0.2, timeout_seconds = 20, network_id: 'base-sepolia')
```
[the remark this answers is not in the document]
Comment (jazz-cb) [reply lib/coinbase/wallet.rb]: done

File: lib/coinbase/wallet.rb [hunk 3]
```diff
@@ -36,10 +47,14 @@ def import(data)
       end
 
       # Creates a new Wallet on the specified Network and generate a default address for it.
+      # @param interval_seconds [Integer] The interval at which to poll the CDPService for the Wallet to
```
Comment (yuga-cb) [inline lib/coinbase/wallet.rb:50]: Nit: change the order of these

**Review by yuga-cb — APPROVED** [review]

File: lib/coinbase/wallet.rb (hunk 3 above)
Comment (jazz-cb) [reply lib/coinbase/wallet.rb:50]: will fix in the next PR 👍

**Pull request merged**